lib/msf/core/payload/windows/migrate_pipe.rb

```diff
@@ -0,0 +1,59 @@
+# -*- coding: binary -*-
+
+require 'msf/core'
+require 'msf/core/payload/windows/block_api'
+
+module Msf
+
+###
+#
+# Payload that supports migrating over Named Pipe transports on x86.
+#
+###
+
+module Payload::Windows::MigratePipe
+
+  include Msf::Payload::Windows
+  include Msf::Payload::Windows::BlockApi
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'Migrate over Named Pipe transport',
+      'Description'   => 'Migration stub to use over Named Pipe transports',
+      'Author'        => ['OJ Reeves'],
+      'License'       => MSF_LICENSE,
+      'Platform'      => 'win',
+      'Arch'          => ARCH_X86,
+    ))
+  end
+
+  #
+  # Constructs the payload
+  #
+  def generate
+    asm = %Q^
+    migrate:
+      cld
+      mov esi, [esp+4]
```

Comment:
This is shorter:
```
pop esi
pop esi
```
A comment here to explain what should be on the stack would be useful.